src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -218,8 +218,11 @@ def __call__(
             text_input_ids = text_input_ids[:, : self.tokenizer.model_max_length]
         text_embeddings = self.text_encoder(text_input_ids.to(self.device))[0]
 
-        # duplicate text embeddings for each generation per prompt
-        text_embeddings = text_embeddings.repeat_interleave(num_images_per_prompt, dim=0)
+        # duplicate text embeddings for each generation per prompt, using mps friendly method
+        batch_size, seq_len, _ = text_embeddings.shape
+        text_embeddings = text_embeddings.repeat(1, num_images_per_prompt, 1).view(
+            batch_size * num_images_per_prompt, seq_len, -1
+        )
```

Comment:
(nit) maybe split `repeat` and `view` in multiple lines for better readability.

Reply:
Good idea. I just applied what `make style` thought was ok.